File: tests/fixtures/rust/excess_args.rs
fn create_user(
    first_name: &str,
    last_name: &str,
    email: &str,
    phone: &str,
    address: &str,
    city: &str,
    state: &str,
    zipcode: &str,
) -> String {
    format!("{} {}", first_name, last_name)
}

fn simple_func(a: i32, b: i32) -> i32 {
    a + b
}

struct UserService {
    db: String,
    cache: String,
    logger: String,
    mailer: String,
    validator: String,
    scheduler: String,
}

impl UserService {
    fn new(
        db: String,
        cache: String,
        logger: String,
        mailer: String,
        validator: String,
        scheduler: String,
    ) -> Self {
        Self { db, cache, logger, mailer, validator, scheduler }
    }

    fn get_user(&self, user_id: &str) -> Option<String> {
        Some(self.db.clone())
    }
}

File: tests/fixtures/csharp/excess_args.cs
public class UserManager
{
    public object CreateUser(string firstName, string lastName, string email,
                             string phone, string address, string city,
                             string state, string zipcode)
    {
        return new { firstName, lastName, email, phone, address, city, state, zipcode };
    }

    public int SimpleFunc(int a, int b)
    {
        return a + b;
    }
}

public class UserService
{
    private object db;
    private object cache;
    private object logger;
    private object mailer;
    private object validator;
    private object scheduler;

    public UserService(object db, object cache, object logger,
                       object mailer, object validator, object scheduler,
                       object extra7, object extra8, object extra9)
    {
        this.db = db;
        this.cache = cache;
        this.logger = logger;
        this.mailer = mailer;
        this.validator = validator;
        this.scheduler = scheduler;
    }

    public object GetUser(string userId)
    {
        return null;
    }
}
